File: make_icon.js
/**
 * make_icon.js — Renders Claude logo SVG to PNG using sharp + svg2img
 * Run: node make_icon.js
 * Requires: npm install sharp
 */
const fs   = require('fs');
const path = require('path');
const os   = require('os');
const { execSync } = require('child_process');

const OUT = path.join(__dirname, 'resources');
fs.mkdirSync(OUT, { recursive: true });

// Find Claude Code SVG
const extRoot = path.join(os.homedir(), '.vscode', 'extensions');
const claudeDir = fs.readdirSync(extRoot).find(d => d.startsWith('anthropic.claude-code-'));
const svgPath = path.join(extRoot, claudeDir, 'resources', 'claude-logo.svg');
const svgOrig = fs.readFileSync(svgPath, 'utf8');

// Recolor helpers
const white = svgOrig.replace(/#D97757/g, 'white').replace(/width="1em"/, 'width="200"').replace(/height="1em"/, 'height="200"');
const black = svgOrig.replace(/#D97757/g, 'black').replace(/width="1em"/, 'width="200"').replace(/height="1em"/, 'height="200"');

fs.writeFileSync(path.join(OUT, '_tmp_white.svg'), white);
fs.writeFileSync(path.join(OUT, '_tmp_black.svg'), black);

// Check sharp
try { require.resolve('sharp'); } catch(e) {
    console.log('Installing sharp...');
    execSync('npm install sharp', { cwd: __dirname, stdio: 'inherit' });
}
const sharp = require('sharp');

async function main() {
    // dark theme: white logo, transparent bg
    await sharp(Buffer.from(white), { density: 300 })
        .resize(32, 32)
        .png()
        .toFile(path.join(OUT, 'icon-dark.png'));
    console.log('icon-dark.png done');

    // light theme: black logo, transparent bg
    await sharp(Buffer.from(black), { density: 300 })
        .resize(32, 32)
        .png()
        .toFile(path.join(OUT, 'icon-light.png'));
    console.log('icon-light.png done');

    // marketplace: white logo on black circle, 128x128
    const logoWhite = await sharp(Buffer.from(white), { density: 300 })
        .resize(100, 100)
        .png()
        .toBuffer();

    const circle = Buffer.from(`<svg width="128" height="128" xmlns="http://www.w3.org/2000/svg">
        <circle cx="64" cy="64" r="64" fill="black"/>
    </svg>`);

    await sharp(circle)
        .composite([{ input: logoWhite, gravity: 'center' }])
        .png()
        .toFile(path.join(OUT, 'icon.png'));
    console.log('icon.png done');

    // cleanup
    fs.unlinkSync(path.join(OUT, '_tmp_white.svg'));
    fs.unlinkSync(path.join(OUT, '_tmp_black.svg'));
    console.log('All done.');
}

main().catch(e => { console.error(e); process.exit(1); });

File: make_icon.py
"""
make_icon.py -- Generate black edition icons for Claude Codex Black Ed.
The Claude logo is a filled circle with an internal asterisk pattern.
We desaturate + boost contrast to reveal that pattern, then invert for themes.

Output:
  resources/icon.png       -- 128x128 black circle, white asterisk (extension panel)
  resources/icon-dark.png  -- 32x32 for dark VSCode theme (white logo on transparent)
  resources/icon-light.png -- 32x32 for light VSCode theme (black logo on transparent)

Run: python make_icon.py
"""

import subprocess, shutil, sys, os

SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
RESOURCES  = os.path.join(SCRIPT_DIR, 'resources')
CACHED_SRC = os.path.join(RESOURCES, 'claude-logo-source.png')

EXT_ROOT = os.path.join(os.path.expanduser('~'), '.vscode', 'extensions')
os.makedirs(RESOURCES, exist_ok=True)

def findSource():
    if os.path.exists(CACHED_SRC):
        return CACHED_SRC
    if os.path.isdir(EXT_ROOT):
        for d in sorted(os.listdir(EXT_ROOT), reverse=True):
            if d.startswith('anthropic.claude-code-'):
                p = os.path.join(EXT_ROOT, d, 'resources', 'claude-logo.png')
                if os.path.exists(p):
                    shutil.copy2(p, CACHED_SRC)
                    print(f'  Cached source -> {CACHED_SRC}')
                    return CACHED_SRC
    raise RuntimeError('Claude logo not found. Install Claude Code extension first.')

def findMagick():
    for cmd in ['magick', 'convert']:
        if shutil.which(cmd): return cmd
    raise RuntimeError('ImageMagick not found. Install: choco install imagemagick')

def run(args, label):
    print(f'  {label}...')
    r = subprocess.run(args, capture_output=True, text=True)
    if r.returncode != 0:
        print(f'  ERROR: {r.stderr.strip()}')
        sys.exit(1)
    print('  OK')

def main():
    magick = findMagick()
    src    = findSource()
    print(f'ImageMagick: {magick}')
    print(f'Source:      {src}')

    out128 = os.path.join(RESOURCES, 'icon.png')
    out32d = os.path.join(RESOURCES, 'icon-dark.png')
    out32l = os.path.join(RESOURCES, 'icon-light.png')

    # Shared processing chain:
    # 1. Flatten onto white (removes transparency artifacts)
    # 2. Desaturate to reveal internal contrast (asterisk vs circle)
    # 3. Boost contrast so the asterisk pops clearly
    # 4. Invert: dark circle with white asterisk
    PROCESS = [
        '-background', 'white', '-flatten',
        '-colorspace', 'Gray',
        '-sigmoidal-contrast', '8,50%',   # strong S-curve to boost contrast
        '-negate',                         # invert: circle=black, asterisk=white
    ]

    # 128x128 for extension panel (black circle, white asterisk, white bg)
    run([magick, src, '-resize', '128x128'] + PROCESS + [out128],
        f'128x128 -> {out128}')

    # 32x32 dark theme: white circle, black asterisk, transparent bg
    # We need to put it on transparent, not white
    # Process on white, then make white transparent
    run([
        magick, src, '-resize', '32x32',
        '-background', 'white', '-flatten',
        '-colorspace', 'Gray',
        '-sigmoidal-contrast', '8,50%',
        # DON'T invert: dark circle on white -> white bg becomes transparent
        '-negate',                          # black circle, white asterisk
        '-negate',                          # back: white circle, black asterisk
        # For dark theme we want LIGHT logo on transparent
        # Make bg (white) transparent, then invert colors
        '-fuzz', '5%', '-transparent', 'white',  # remove white bg
        '-negate',                                  # invert remaining: dark->light
        out32d
    ], f'32x32 dark theme -> {out32d}')

    # 32x32 light theme: black circle, white asterisk, transparent bg
    run([
        magick, src, '-resize', '32x32',
        '-background', 'white', '-flatten',
        '-colorspace', 'Gray',
        '-sigmoidal-contrast', '8,50%',
        '-negate',                                  # black circle, white asterisk
        '-fuzz', '5%', '-transparent', 'white',    # make white transparent
        out32l
    ], f'32x32 light theme -> {out32l}')

    print(f'\nDone. Icons in: {RESOURCES}')

if __name__ == '__main__':
    main()
